### ISSUE TITLE
certificate error when running from npm run script

### ISSUE THREAD

**Issue by agaengel** [tjunnone/npm-check-updates#604]
I have a project with ncu as dev dependecy that should be run as npm run ncu on our jenkins.

but script looks like this:
"scripts": {
"ncu": "node_modules/npm-check-updates/bin/ncu --loglevel verbose --packageFile package.json"
} 

the script fails with the following error :
```
/opt/jenkins/workspace/test_artifactory/node_modules/npm-check-updates/lib/npm-check-updates.js:392
    throw err;
    ^
FetchError: request to https://artifactory.local/artifactory/api/npm/npm-virtual/@vue%2fcli-plugin-eslint failed, reason: self signed certificate in certificate chain
    at ClientRequest.req.on.err (/opt/jenkins/workspace/test_artifactory/node_modules/node-fetch-npm/src/index.js:68:14)
    at ClientRequest.emit (events.js:198:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
 
npm is configured to use a ca with the needee certificate for the artifactory. This works for npm without error

When calling node_modules/npm-check-updates/bin/ncu --loglevel verbose --packageFile package.json directly without npm run this also works.

The only workaround for the jenkins is adding strict-ssl=false to npmrc then also the npm run ncu works

is ncu not useing the certificate from the configured npmrc ?

Regards
Alex

* node version: v10.16.3
* npm version: 6.9.0
* npm-check-updates version: 3.2.0

---------------------------

**Comment by raineorshine** [comment]
I'm quite unfamiliar with certificates in npm, but as a guess, is this article relevant?
https://medium.com/@jonatascastro12/understanding-self-signed-certificate-in-chain-issues-on-node-js-npm-git-and-other-applications-ad88547e7028

**Comment by agaengel** [comment]
the ca is set via cafile in the npmrc and this works as long as I run this directly but with npm run this fails, so it might be something inside npm run

**Comment by raineorshine** [comment]
Can you confirm that setting NODE_EXTRA_CA_CERTS in the npm run script does not work?

```
scripts": {
  "ncu": "NODE_EXTRA_CA_CERTS=/path/to/your/cert.pem ncu --loglevel verbose --packageFile package.json"
}
```

**Comment by agaengel** [comment]
using NODE_EXTRA_CA_CERTS in the script  works we set the ca in our like: cafile=/path/to/your/cert.pem

**Comment by stoically** [comment]
My best would be that the cafile isn't reachable at the given position on jenkins, maybe add a `ls /path/to/your/cert.pem` to the pipe to make sure? (ncu, or rather pacote for that matter, will silently ignore not existing ca files)

I've tested with and without `npm run`, and in both cases the npmrc is correctly, and since adding `strict-ssl=false` works in your case, that's also true for the jenkins scenario.

**Comment by raineorshine** [comment]
@agaengel Potential fix made in #713 and published in `v7.0.3-alpha.1`. Will you try it out and let me know if it fixes the problem for you? 

```sh
npm install -g npm-check-updates@next
```
